### PR TITLE
Pick reviewers from outside the bounty first

### PR DESCRIPTION
**Pull request by jtakalai**

### CODE REVIEW

File: packages/network-contracts/test/hardhat/BrokerEconomics/BountyPolicies/VoteKickPolicy.test.ts [hunk 1]
```diff
@@ -130,7 +120,13 @@ describe("VoteKickPolicy", (): void => {
...
     describe("Flagging + reviewer selection", function(): void {
         it("picks first brokers that are not in the same bounty", async () => {
-            // TODO
+            const { bounty, brokers, pools: [ pool1, flaggedPool ] } = await setupBounty(4, 4, "pick-first-nonstaked-brokers")
+            const flagReceipt = await (await bounty.connect(brokers[0]).flag(flaggedPool.address, pool1.address)).wait() as ContractReceipt
+            const reviewers = flagReceipt.events!.filter((e) => e.event === "ReviewRequest").map((e) => e.args?.reviewer)
+            // console.log("Brokers %o", brokers.map((b) => b.address))
+            // console.log("Flagged pool %o", flaggedPool.address)
+            // console.log("Reviewers %o", reviewers)
+            expect(reviewers.slice(0, 4)).to.have.members([brokers[4].address, brokers[5].address, brokers[6].address, brokers[7].address])
```
Comment (jtakalai) [inline packages/network-contracts/test/hardhat/BrokerEconomics/BountyPolicies/VoteKickPolicy.test.ts:129]: here's the gist of this commit: test

File: packages/network-contracts/contracts/BrokerEconomics/BountyPolicies/VoteKickPolicy.sol [hunk 2]
```diff
@@ -34,120 +38,142 @@ contract VoteKickPolicy is IKickPolicy, Bounty {
...
+        // secondary selection: peers from the same bounty
+        for (uint i = 0; i < sameBountyPeerCount && reviewers[target].length < REVIEWER_COUNT; i++) {
+            address peer = sameBountyPeers[i];
+            if (reviewerState[target][peer] != 0) {
+                continue;
+            }
+            reviewerState[target][peer] = 1;
+            emit ReviewRequest(peer, this, target);
+            reviewers[target].push(peer);
+        }
```
Comment (jtakalai) [inline packages/network-contracts/contracts/BrokerEconomics/BountyPolicies/VoteKickPolicy.sol:94]: here's the gist of this commit: feature code

**Review by samt1803 — APPROVED** [review]
looks good

**Review by samt1803 — CHANGES_REQUESTED** [review]
there seems to be an issue with the logic we should go through some tests first

**Review by samt1803 — APPROVED** [review]

**Pull request merged**